### PR TITLE
[Form] Document the `hash_property_path` option

### PR DESCRIPTION
**Pull request by Seb33300**
Documentation for https://github.com/symfony/symfony/pull/46224

### CODE REVIEW

File: reference/forms/types/password.rst [hunk 1]
```diff
@@ -35,6 +35,28 @@ with the ``value`` attribute set to its true value only upon submission.
...
+    To minimize the risk of leaking the plain password,
+    this option can only be used with the ``mapped`` option set to ``false``.
+
+        $builder->add('plainPassword', PasswordType::class, [
+            'hash_property_path' => 'password',
+            'mapped' => false,
+        ]);
+
+    See the :ref:`"mapped" option documentation <reference-form-mapped>`.
+
```
Comment (OskarStark) [inline reference/forms/types/password.rst]:
```suggestion

.. versionadded:: 6.2

    The ``hash_property_path`` option was introduced in Symfony 6.2.

```

File: reference/forms/types/password.rst [hunk 2]
```diff
@@ -35,6 +35,28 @@ with the ``value`` attribute set to its true value only upon submission.
...
+If set, the password will be hashed using the PasswordHasher component and
+stored in the specified property.
+
+Data passed to the form must be a ``PasswordAuthenticatedUserInterface`` object.
+
+.. caution::
+
+    To minimize the risk of leaking the plain password,
+    this option can only be used with the ``mapped`` option set to ``false``.
```
Comment (OskarStark) [inline reference/forms/types/password.rst]:
```suggestion
    this option can only be used with the ``mapped`` option set to ``false``::
```

File: reference/forms/types/password.rst [hunk 3]
```diff
@@ -35,6 +35,28 @@ with the ``value`` attribute set to its true value only upon submission.
 If you want to render your password field *with* the password value already
 entered into the box, set this to false and submit the form.
 
+``hash_property_path``
+~~~~~~~~~~~~~~~~
```
Comment (OskarStark) [inline reference/forms/types/password.rst]: Underline too short

Comment (Seb33300) [reply reference/forms/types/password.rst]: Fixed. Thanks

**Comment by Seb33300** [comment]
This feature is now merged in 6.2 🎉

@javiereguiluz can we merge this PR?

**Comment by wouterj** [comment]
Yes we can! Thanks for providing both feature and docs, @Seb33300!

fyi, I've slightly updated the PR while merging (move the versionadded and add some more links).

**Comment by maxhelias** [comment]
I think, we should also explain how to configure it on a `RepeatedType`

**Comment by Seb33300** [comment]
If you want to create a PR to add it to the documentation:

```php
$builder->add('plainPassword', RepeatedType::class, [
    'type' => PasswordType::class,
    'first_options'  => ['label' => 'Password', 'hash_property_path' => 'password'],
    'second_options' => ['label' => 'Repeat Password'],
    'mapped' => false,
]);
```

**Comment by OskarStark** [comment]
Here you go
* https://github.com/symfony/symfony-docs/pull/17487

**Pull request merged**